bootstrap/src/test/java/io/airlift/bootstrap/TestBootstrap.java

```diff
@@ -81,6 +81,34 @@ public void testNonStrictConfig()
                 .initialize();
     }
 
+    @Test
+    public void testUserErrorsReported()
+    {
+        Bootstrap bootstrap = new Bootstrap(binder -> {
+            throw new RuntimeException("happy user error");
+        });
+
+        assertThatThrownBy(bootstrap::initialize)
+                .isInstanceOfSatisfying(ApplicationConfigurationException.class, e ->
+                        assertThat(e.getErrors().stream().map(Message::getMessage)).containsExactly(
+                                "An exception was caught and reported. Message: happy user error"));
```

Comment:
can we verify that e printout (`Throwables.getStackTraceAsString`) contains the `happy user error`?

inspecting `e.getErrors()` is not how i usually interact with exceptions

Reply:
good point. Added.